lib/jbuilder/jbuilder_template.rb

```diff
@@ -199,6 +199,7 @@ def _render_explicit_partial(name_or_options, locals = {})
     when ::Hash
       # partial! partial: 'name', foo: 'bar'
       options = name_or_options
+      options[:locals] ||= options.except(:partial, :as, :collection)
```

Comment:
If we want to take these out of the original options hash, I think we'd need to do something like
```ruby
options[:locals] ||= options.except(:partial, :as, :collection).keys.each_with_object({}) do |local, hash|
  hash[local] = options.delete(local)
end
```
Thoughts?